### PR TITLE
B-19722-INT USCG Customer must enter EMPLID

### PR DESCRIPTION
**Pull request by paulstonebraker**
## Summary

USCG customers must enter EMPLID on account creation. This value is editable when a move is in draft state.

### How to test

1. Create a new customer
2. Select Coast Guard as branch and enter EMPLID
3. Go back and forth between pages and try to break (change branches, enter letters, etc)
4. If not broken advance through profile and orders creation until you have a move in draft. Access your profile screen and edit your EMPLID and try to break
5. If not broken, create a new customer and confirm that EMPLID must be unique

### CODE REVIEW

File: src/pages/MyMove/Profile/DodInfo.jsx [hunk 1]
```diff
@@ -44,7 +55,13 @@ export const DodInfo = ({ updateServiceMember, serviceMember, oktaUser }) => {
       .catch((e) => {
         // Error shape: https://github.com/swagger-api/swagger-js/blob/master/docs/usage/http-client.md#errors
         const { response } = e;
-        const errorMessage = getResponseError(response, 'failed to update service member due to server error');
+        let errorMessage;
+        if (e.response.body.message === 'Unhandled data error encountered') {
+          errorMessage = 'This EMPLID is already in use';
```
Comment (paulstonebraker) [inline src/pages/MyMove/Profile/DodInfo.jsx:60]: @danieljordan-caci any thoughts on handling the error this way

Comment (danieljordan-caci) [reply src/pages/MyMove/Profile/DodInfo.jsx:60]: I don't hate it, but part of me doesn't like the specific message based off of a generalized server error response. I haven't had a chance to pull this and test it, but initial gut response is... It works, yeah, but can the return error be more specific to the error you display to the user?

Comment (paulstonebraker) [reply src/pages/MyMove/Profile/DodInfo.jsx:60]: Should've provided more details. So this error gets returned in lines 182-184 of service_members.go, I followed the error handling to errors.go `ResponseForVErrors` and `ResponseForError` where it returns that message on line 176 `case *pq.Error`

I did a quick search of the internal app and didn't see much in the way of handling an error more specifically. So... I couldn't find any code to copy, which is why I tagged you 🤣

File: src/components/Customer/Review/ServiceInfoDisplay/ServiceInfoDisplay.jsx [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
 }) => {
   const { state } = useLocation();
+  const [isEmplidEnabled, setIsEmplidEnabled] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsEmplidEnabled(await isBooleanFlagEnabled('coast_guard_emplid'));
+    };
+    fetchData();
+  });
```
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply src/components/Customer/Review/ServiceInfoDisplay/ServiceInfoDisplay.jsx:30]: Then you wouldn't need to use any hooks in the `ServiceInfoDisplay` component.

File: src/pages/MyMove/Profile/DodInfo.jsx [hunk 3]
```diff
@@ -13,14 +13,24 @@ import requireCustomerState from 'containers/requireCustomerState/requireCustome
...
   const navigate = useNavigate();
   const [serverError, setServerError] = useState(null);
+  const [isEmplidEnabled, setIsEmplidEnabled] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsEmplidEnabled(await isBooleanFlagEnabled('coast_guard_emplid'));
+    };
+    fetchData();
+  });
```
Comment (danieljordan-caci) [inline src/pages/MyMove/Profile/DodInfo.jsx:28]: Same thing here - adding an empty array so React knows to only run this once.

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Please make sure you're also updating the storybooks.

File: src/components/Customer/Review/ServiceInfoDisplay/ServiceInfoDisplay.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulstonebraker) [reply src/components/Customer/Review/ServiceInfoDisplay/ServiceInfoDisplay.jsx:30]: Nicely done, thanks. WIll update using the latter shortly

**Comment by paulstonebraker** [comment]
Converting to draft while I make storybook changes

File: src/components/Customer/Review/ServiceInfoDisplay/ServiceInfoDisplay.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulstonebraker) [reply src/components/Customer/Review/ServiceInfoDisplay/ServiceInfoDisplay.jsx]: Implemented all this except for the `Loading` component as we don't seem to use that elsewhere with the feature flags

**Comment by danieljordan-caci** [comment]
Conflicts

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Thanks for making those changes and updating storybook. 

LGTM.

**Review by danieljordan-caci — APPROVED** [review]

**Review by pambecker — APPROVED** [review]
good to go once you create a PR for the milmove-feature-flags side of things

**Pull request merged**